src/coord/timestamp.rs

```diff
@@ -244,6 +340,109 @@ pub struct Timestamper {
     max_increment_size: i64,
 }
 
+/// A debezium record contains a set of update counts for each topic that the transaction
+/// updated. This function extracts the set of (topic, update_count) as a vector.
+fn parse_debezium(record: Vec<(String, Value)>) -> Vec<(String, i64)> {
+    let mut result = vec![];
+    for (key, value) in record {
+        if key == "data_collections" {
+            if let Value::Union(_, value) = value {
+                if let Value::Array(items) = *value {
+                    for v in items {
+                        if let Value::Record(item) = v {
+                            let mut value: String = String::new();
+                            let mut write_count = 0;
+                            for (k, v) in item {
+                                if k == "data_collection" {
+                                    if let Value::String(data) = v {
+                                        value = data;
+                                    } else {
+                                        error!("Incorrect AVRO format. String expected");
+                                    }
+                                } else if k == "event_count" {
+                                    if let Value::Long(e) = v {
+                                        write_count = e;
+                                    } else {
+                                        error!("Incorrect AVRO format. Long expected");
+                                    }
+                                }
+                            }
+                            result.push((value, write_count))
+                        } else {
+                            error!("Incorrect AVRO format. Record expected");
+                        }
+                    }
+                }
+            } else {
+                error!(
+                    "Incorrect AVRO format. Union of Null/Array expected {:?}",
+                    value
+                );
+            }
+        }
+    }
+    result
+}
+/// Determines what format does a given consistency source abide to
+/// TODO(Natacha): this information should be included as metadata
+/// This function tries to decode to all known formats. If a decoding
+/// is successful, it assumes that all subsequent messages of that stream
+/// will have been encoded using the same formatting/envelope
+fn identify_consistency_format(msgs: &[Vec<u8>]) -> ConsistencyFormatting {
+    if let Some(msg) = msgs.first() {
+        let mut bytes = &msg[5..];
+        let res = futures::executor::block_on(avro::from_avro_datum(
```

Comment:
/cc @umanwizard are you still planning to un-asyncify `from_avro_datum`?

Reply:
yes but you can just leave this as is for now, I will fix all the callsites